package/MDAnalysis/converters/RDKit.py

```diff
@@ -693,27 +774,30 @@ def _run_reaction(reaction, reactant):
     product : rdkit.Chem.rdchem.Mol
         The final product of the reaction
     """
-    # count how many times the reaction should be run
-    pattern = Chem.MolFromSmarts(reaction.split(">>")[0])
-    n_matches = len(reactant.GetSubstructMatches(pattern))
-
-    # run the reaction for each matched pattern
-    rxn = AllChem.ReactionFromSmarts(reaction)
-    for n in range(n_matches):
-        products = rxn.RunReactants((reactant,))
-        # only keep the first product
+    # repeat the reaction until all matching moeities have been transformed
+    # note: this loop is meant to be ended by a `break` statement
+    # but let's avoid using `while` loops just in case
+    for n in range(reactant.GetNumAtoms()):
+        reactant.UpdatePropertyCache(strict=False)
+        Chem.Kekulize(reactant)
+        products = reaction.RunReactants((reactant,))
```

Comment:
Would this be a good spot for `RunReactantInPlace`? Would that then save the property updating?

Reply:
Yes, although it's been introduced in RDKit v2021.09.1 so I'm not sure if it's a good idea to implement it in the code right now.  
It will very likely speed things up a ton (and simplify the codebase), but we can focus on optimization later I think ?